src/basic/namespace-util.c

```diff
@@ -48,6 +49,62 @@ NamespaceType clone_flag_to_namespace_type(unsigned long clone_flag) {
         return _NAMESPACE_TYPE_INVALID;
 }
 
+static int pidref_namespace_open_by_type_internal(const PidRef *pidref, NamespaceType type, bool *need_verify) {
+        int r;
+
+        assert(pidref_is_set(pidref));
+        assert(type >= 0 && type < _NAMESPACE_TYPE_MAX);
+
+        if (pidref_is_remote(pidref))
+                return -EREMOTE;
+
+        if (pidref->fd >= 0) {
+                r = pidfd_get_namespace(pidref->fd, namespace_info[type].pidfd_get_ns_ioctl_cmd);
+                if (r != -EOPNOTSUPP)
```

Comment:
given that you now return a recognizable ENOPKG error for the other kind of "not supported", you could just use ERRNO_IS_NOT_SUPPORTED() here, just to make this less surprising.

(or to say this differently: if we do not use ERRNO_IS_NOT_SUPPORTED() then we really should add a comment why not. But there's no reason to not use it here, hence we should)